include/courtier/GWebsocketConsumerT.hpp

```diff
@@ -1508,6 +1513,445 @@ class GWebsocketConsumerT
+	 /**
+	  * Allos to configure the serialization mode for the communication between
+	  * clients and server
+	  *
+	  * @param serializationMode The serialization mode to be configured for this class
+	  */
+	 void setSerializationMode(Gem::Common::serializationMode serializationMode) {
+	 	 m_serializationMode = serializationMode;
+	 }
+
+	 //-------------------------------------------------------------------------
+	 /**
+	  * Allows to retrieve the serialization mode configured for this class
+	  *
+	  * @return The serialization mode configured for this class
+	  */
+	 Gem::Common::serializationMode getSerializationMode() const {
+	 	return m_serializationMode;
+	 } 
+
+ 	 //-------------------------------------------------------------------------
+	 /**
+	  * Configures the number of threads to be used by this class
+	  *
+	  * @param The number of threads to be used by this class
+	  */
+	 void setNThreads(std::size_t nThreads) {
+		 // Adapt the number of processing threads, if automatic detection was requested
+		 if(0 == nThreads) {
+			 glogger
+				 << "In GWebsocketConsumerPT<>::setNThreads(): " << std::endl
+				 << "nThreads was set to 0. m_n_threads will be set to default " << GCONSUMERLISTENERTHREADS << std::endl
+				 << "This replaces the old behaviour where a value of 0 would have" << std::endl
+				 << "resulted in the number of hardware threads being unsed" << std::endl
+				 << GWARNING;
+
+			 m_n_listener_threads = GCONSUMERLISTENERTHREADS;
+		 } else {
+			 m_n_listener_threads = nThreads;
+		 }
+	 }
+
+	 //-------------------------------------------------------------------------
+	 /**
+	  * Allows to retrieve the number of processing threads to be used for processing
+	  * incoming connections in the server
+	  */
+  	 std::size_t getNProcessingThreads() const {
+  	 	return m_n_listener_threads;
+  	 }
+
+
+  
+protected:
+	 //-------------------------------------------------------------------------
+	 /**
+	  * Stop execution
+	  */
+	 void shutdown_() override {
+		 //------------------------------------------------------
+		 // Set the class-wide shutdown-flag
+		 GBaseConsumerT<processable_type>::shutdown_();
+
+		 // Make sure context threads may terminate
+		 m_io_contexts.stop();
+
+	 }
+
+private:
+	 //-------------------------------------------------------------------------
+	 /**
+	  * Adds local command line options to a boost::program_options::options_description object.
+	  *
+	  * @param visible Command line options that should always be visible
+	  * @param hidden Command line options that should only be visible upon request
+	  */
+	 void addCLOptions_(
+		 boost::program_options::options_description &visible
+		 , boost::program_options::options_description &hidden
+	 ) override {
+		 namespace po = boost::program_options;
+
+		 visible.add_options()
+			 ("beast_ip", po::value<std::string>(&m_server)->default_value(GCONSUMERDEFAULTSERVER),
```

Comment:
This seems to duplicate the command line settings from GWebsocketConsumerT. It is not clear to me whether this class is meant as an alternative or a replacement?